File: Selection Sort/sort.py
import random

def findSmallest(arr):
    smallest = arr[0]
    smallestIndex = 0
    
    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallestIndex = i
    return smallestIndex

def selectionSort(arr):
    newArr = []
    for _ in range(len(arr)):
        smallestIndex = findSmallest(arr)
        newArr.append(arr.pop(smallestIndex))
    return newArr

myList = []

for _ in range(50):
    num = random.randint(1, 1000)
    myList.append(num)

print(f"Original list: {myList}")
sortedList = selectionSort(myList.copy())  
print(f"Sorted list: {sortedList}")

File: Selection Sort/sort.js
function findSmallest(arr) {
    let smallest = arr[0];
    let smallestIndex = 0;
    
    for (let i = 1; i < arr.length; i++) {
        if (arr[i] < smallest) {
            smallest = arr[i];
            smallestIndex = i;
        }
    }
    return smallestIndex;
}

function selectionSort(arr) {
    let newArr = [];
    const len = arr.length;
    for (let i = 0; i < len; i++) {
        let smallestIndex = findSmallest(arr);
        newArr.push(arr.splice(smallestIndex, 1)[0]);
    }
    return newArr;
}

let myList = [];

for (let i = 0; i < 50; i++) {
    let num = Math.floor(Math.random() * 1000) + 1;
    myList.push(num);
}

console.log("Original list:");
console.log(myList);
let sortedList = selectionSort([...myList]);  // Make a copy of the original list to preserve the original order
console.log("Sorted list:");
console.log(sortedList);
